File: projekt_1/Projekt_1/Projekt_1/Wilk.cpp
#pragma once
#include "Wilk.h"
using namespace std;

Wilk::Wilk(Swiat* swiat, int x, int y) : Zwierze(swiat, x, y, WILK_SILA, WILK_INIC, WILK_PELNOLETNOSC) {}

bool Wilk::Koliduj(Organizm* somsiad, bool czyAtakuje) {
	return false;
}

void Wilk::dodajPolog() { this->polog = WILK_POLOG; }
Organizm* Wilk::dajPotomka(Wsp n_wsp) {
	Organizm* tmp = new Wilk(swiat, n_wsp.x, n_wsp.y);
	swiat->DodajOrganizm(tmp);
	return tmp;
}
bool Wilk::czyTenSamGatunek(Organizm* o) { if (o->getChar() == WILK_CHAR) { return true; } else return false; }

char Wilk::getChar() { return WILK_CHAR; }
string Wilk::getNazwa() { return WILK_NAZWA; }

File: projekt_2/gra/resources/Zwierzeta/Wilk.java
package gra.resources.Zwierzeta;

import gra.resources.backend.*;

public class Wilk extends Zwierze{
    private final static int WILK_SILA = 9;
    private final static int WILK_INIC = 5;
    private final static char WILK_CHAR = 'W';
    private final static int WILK_POLOG = 5;
    private final static int WILK_PELNOLETNOSC = 10;
    private final static String WILK_NAZWA = "Wilk";

    public Wilk(Swiat swiat, int x, int y) { super(swiat, x, y, WILK_SILA, WILK_INIC, WILK_PELNOLETNOSC); }

    public boolean Koliduj(Organizm somsiad, boolean czyAtakuje) {
        return false;
    }
    public void dodajPolog() { this.polog = WILK_POLOG; }
    public Organizm dajPotomka(Wsp n_wsp) {
        Organizm tmp = new Wilk(swiat, n_wsp.x, n_wsp.y);
        swiat.DodajOrganizm(tmp);
        return tmp;
    }
    public boolean czyTenSamGatunek(Organizm o) { if (o instanceof Wilk) { return true; } else { return false; } }
    public char getChar() { return WILK_CHAR; }
    public String getNazwa() { return WILK_NAZWA; }

}
